doc/man7/EVP_KDF-ARGON2.pod

```diff
@@ -0,0 +1,150 @@
+=pod
+
+=head1 NAME
+
+EVP_KDF-ARGON2 - The Argon2 EVP KDF implementation
+
+=head1 DESCRIPTION
+
+Support for computing the B<argon2> password-based KDF through the B<EVP_KDF>
+API.
+
+The EVP_KDF-ARGON2 algorithm implements the Argon2 password-based key
+derivation function, as described in IETF RFC 9106.  It is memory-hard in
+the sense that it deliberately requires a significant amount of RAM for efficient
+computation. The intention of this is to render brute forcing of passwords on
+systems that lack large amounts of main memory (such as GPUs or ASICs)
+computationally infeasible.
+
+=head2 Supported parameters
+
+The supported parameters are:
+
+=over 4
+
+=item "pass" (B<OSSL_KDF_PARAM_PASSWORD>) <octet string>
+
+=item "salt" (B<OSSL_KDF_PARAM_SALT>) <octet string>
+
+=item "secret" (B<OSSL_KDF_PARAM_SECRET>) <octet string>
+
+=item "iter" (B<OSSL_KDF_PARAM_ITER>) <unsigned integer>
+
+These parameters work as described in L<EVP_KDF(3)/PARAMETERS>.
+
+=item "t" (B<OSSL_KDF_PARAM_THREADS>) <unsigned integer>
+
+The number of threads. This can only be used with built-in thread support.
+Threading must be explicitly enabled. See EXAMPLES section for more information.
+
+=item "s" (B<OSSL_KDF_PARAM_ARGON2_SZ>) <unsigned integer>
+
+Digest length.
+
+=item "a" (B<OSSL_KDF_PARAM_ARGON2_AD>) <octet string>
+
+Optional associated data.
```

Comment:
Maybe add some more description to what this is used for.

Reply:
Would something like `Optional associated data, may be used to "tag" a group of keys, or tie them to a particular public key, without having to modify salt.` work?

Reply:
Yeah that would WFM. Please also note there is a bug in the param name - it is `ad` not `a`.